src/Subscriber/TranslationsSubscriber.php

```diff
@@ -69,10 +85,17 @@ public static function get_subscribed_events()
      */
     public function enforce_locale($locale, $domain)
     {
-        if ($domain == $this->domain) {
-            $locale = 'en_US';
+
+        if ($domain !== $this->domain) {
+            return $locale;
+        }
+        if ( $locale === 'en_US') {
+            return $locale;
         }
 
+        if( !is_readable( $this->mofile ) ){
+            $locale = 'en_US';
+        }
```

Comment:
You're right that there's a design issue with this. But this isn't quite the way to solve it. I'm going to create a fix for it and you can test it.